### PR TITLE
Store: Add image and description fields to product category management

### PR DESCRIPTION
**Pull request by justinshreve**
This PR continues building out the product category management form (see #20295) by adding a description field, and image management.

To Test:
* Boot up this branch and visit `http://calypso.localhost:3000/store/products/categories/:store`.
* Select a category with an image and description (create one via wp-admin if you need to)
* Make sure that the image loads, and the description ends up in the TinyMCE instance.
* Verify you can type in the description field.
* Remove the image using the "x".
* Add a new image and verify successful upload.
* Test again by going back to the category listing and selecting the "Add category" button and testing in the create context.
* You can also monitor these updates via the Redux dev tools, to make sure edits state is being properly updated.

Note: Saving has not been hooked up yet.

screen shot 2018-01-08 at 2 36 33 pm

screen shot 2018-01-08 at 2 38 00 pm

### CODE REVIEW

File: client/extensions/woocommerce/app/product-categories/form.js [hunk 1]
```diff
@@ -54,10 +205,19 @@ class ProductCategoryForm extends Component {
...
 				<Card>
-					<FormFieldSet>
-						<FormLabel htmlFor="name">{ translate( 'Category name' ) }</FormLabel>
-						<FormTextInput id="name" value={ category.name || '' } onChange={ this.setName } />
-					</FormFieldSet>
+					<div className="product-categories__form-info-fields">
+						<div className="product-categories__form-image-wrapper">{ this.renderImage() }</div>
+						<div className="product-categories__form-name-description">
+							<FormFieldSet>
+								<FormLabel htmlFor="name">{ translate( 'Category name' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/product-categories/form.js]: Hi! I've found a possible matching string that has already been translated **64** times:
`translate( 'Category Name' )` *ES Score: 13*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-product-cat-form-fields)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/components/product-image-uploader/index.js [hunk 2]
```diff
@@ -202,13 +202,16 @@ class ProductImageUploader extends Component {
 
 	renderUploader() {
 		const { translate, multiple } = this.props;
+
+		const addString = multiple ? translate( 'Add images' ) : translate( 'Add image' );
```
Comment (a8ci18n) [inline client/extensions/woocommerce/components/product-image-uploader/index.js]: Hi! I've found a possible matching string that has already been translated **25** times:
`translate( 'Add Image' )` *ES Score: 11*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-product-cat-form-fields)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/product-categories/form.js [hunk 3]
```diff
@@ -28,12 +34,139 @@ class ProductCategoryForm extends Component {
...
 		editProductCategory( siteId, category, { name } );
 	};
 
+	setDescription = description => {
+		const { siteId, category, editProductCategory } = this.props;
+		editProductCategory( siteId, category, { description } );
+	};
+
+	onSelect = files => {
+		const file = head( files );
```
Comment (timmyc) [inline client/extensions/woocommerce/app/product-categories/form.js]: TIL about _.head, thanks!

File: client/extensions/woocommerce/app/product-categories/form.js [hunk 4]
```diff
@@ -28,12 +34,139 @@ class ProductCategoryForm extends Component {
...
+					<ImagePreloader
+						src={ src }
+						placeholder={ ( placeholder && <img src={ placeholder } /> ) || <span /> }
+					/>
+				</figure>
+			);
+		} else if ( isUploading ) {
+			image = (
+				<figure>
+					<img src={ placeholder || <span /> } />
```
Comment (timmyc) [inline client/extensions/woocommerce/app/product-categories/form.js]: A `<span />` as the img src?

Comment (justinshreve) [reply client/extensions/woocommerce/app/product-categories/form.js]: Doh, thanks. A span is used as a placeholder for `<ImagePreloader>` a few lines up, but yeah -- we shouldn't have that here. Removed in https://github.com/Automattic/wp-calypso/pull/21347/commits/f70504b79f35f0e1f58c2cf0021aff01f381dd98.

**Pull request merged**